### ISSUE TITLE
Cannot build in Windows - your readme file needs updated

### ISSUE THREAD

**Issue by robault** [FLIF-hub/FLIF#279]
I had VS2013 Pro installed but saw VS2015 Community was required so I downloaded, installed it using all default options and rebooted. When I double click on the ".\build\MSVC\dl_make_vs.bat" I get an error indicating a required file is missing. 

'"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools....\VC\vcvarsall.bat"' is not recognized as an internal or external command,
operable program or batch file.
need VS2015
Microsoft Windows [Version 10.0.10586](c) 2015 Microsoft Corporation. All rights reserved.

I searched for "vcvarsall" in "C:\Program File (x86)" and find two, one in in the Microsoft Visual Studio 11.0 and Microsoft Visual Studio 12.0 directories. It looks like the Visual C++ Toolset needs to be installed which is not part of the default install of VS2015 Community. You might want to add that into your readme file.

**Comment by robault** [comment]
Not wanting to waste time tracking down which features to enable, I reinstalled VS2015 with all options and was able to run the batch file. It stopped at another error when it tries to open  the output.png file indicating no program is associated with it. From Explorer I can click on the PNG which automatically opens it in Paint. Here's the error:

        if exist output.png  del output.png
        flif output.flif output.png
        if exist "test-interface.exe" test-interface.exe ..\tmp-test\dummy.flif
interface test has succeeded.
        start output.png
The system cannot execute the specified program.
NMAKE : fatal error U1077: 'start' : return code '0x1'
Stop.

Just trying this out, I could run "mspaint output.png" from the command window and it opened fine. Might want to change that too. At this point, error or not, I have the libflif.dll file in the ".\build\MSVC\" directory now.

**Comment by Enelar** [comment]
+1 for fedora
A lot of errors for SDL, while i am building only flif command line

**Comment by jonsneyers** [comment]
Does `make flif` result in errors about SDL?

**Comment by Enelar** [comment]
@jonsneyers yes

`make flif` causing

```
cd src && make
make[1]: Entering directory '/home/kberezin/code/FLIF/src'
make[1]: sdl2-config: Command not found
make[1]: sdl2-config: Command not found
cc -std=gnu11 -O3 -ggdb3  -Wall -Ilibrary/ viewflif.c -L. -lflif  -o viewflif
viewflif.c:37:1: error: unknown type name ‘SDL_Window’
 SDL_Window* window = NULL;
 ^~~~~~~~~~
viewflif.c:38:1: error: unknown type name ‘SDL_DisplayMode’
 SDL_DisplayMode dm;
 ^~~~~~~~~~~~~~~
viewflif.c:39:1: error: unknown type name ‘SDL_Renderer’
 SDL_Renderer* renderer = NULL;
 ^~~~~~~~~~~~
viewflif.c:40:1: error: unknown type name ‘SDL_Texture’
 SDL_Texture** image_frame = NULL;
 ^~~~~~~~~~~
viewflif.c: In function ‘draw_image’:
viewflif.c:63:9: warning: implicit declaration of function ‘SDL_QueryTexture’ [-Wimplicit-function-declaration]
     if (SDL_QueryTexture(image_frame[frame], NULL, NULL, &ir.w, &ir.h)) { printf("Error: Could not query texture\n"); return; };
         ^~~~~~~~~~~~~~~~
viewflif.c:67:5: warning: implicit declaration of function ‘SDL_GetWindowSize’ [-Wimplicit-function-declaration]
     SDL_GetWindowSize(window, &wr.w, &wr.h);
     ^~~~~~~~~~~~~~~~~
viewflif.c:89:23: warning: implicit declaration of function ‘SDL_RenderClear’ [-Wimplicit-function-declaration]
     if (tr.x || tr.y) SDL_RenderClear(renderer);
                       ^~~~~~~~~~~~~~~
viewflif.c:91:5: warning: implicit declaration of function ‘SDL_RenderCopy’ [-Wimplicit-function-declaration]
     SDL_RenderCopy(renderer, image_frame[frame], NULL, &tr);
     ^~~~~~~~~~~~~~
viewflif.c:93:5: warning: implicit declaration of function ‘SDL_RenderPresent’ [-Wimplicit-function-declaration]
     SDL_RenderPresent(renderer);
     ^~~~~~~~~~~~~~~~~
viewflif.c: In function ‘do_event’:
viewflif.c:98:19: error: ‘SDL_WINDOWEVENT’ undeclared (first use in this function)
     if (e.type == SDL_WINDOWEVENT && e.window.event == SDL_WINDOWEVENT_CLOSE) {printf("Closed\n"); quit=1; return 0;}
                   ^~~~~~~~~~~~~~~
viewflif.c:98:19: note: each undeclared identifier is reported only once for each function it appears in
viewflif.c:98:39: error: ‘SDL_Event {aka union SDL_Event}’ has no member named ‘window’
     if (e.type == SDL_WINDOWEVENT && e.window.event == SDL_WINDOWEVENT_CLOSE) {printf("Closed\n"); quit=1; return 0;}
                                       ^
viewflif.c:98:56: error: ‘SDL_WINDOWEVENT_CLOSE’ undeclared (first use in this function)
     if (e.type == SDL_WINDOWEVENT && e.window.event == SDL_WINDOWEVENT_CLOSE) {printf("Closed\n"); quit=1; return 0;}
                                                        ^~~~~~~~~~~~~~~~~~~~~
viewflif.c:101:39: error: ‘SDL_Event {aka union SDL_Event}’ has no member named ‘window’
     if (e.type == SDL_WINDOWEVENT && e.window.event == SDL_WINDOWEVENT_SIZE_CHANGED && renderer) { draw_image(); }
                                       ^
viewflif.c:101:56: error: ‘SDL_WINDOWEVENT_SIZE_CHANGED’ undeclared (first use in this function)
     if (e.type == SDL_WINDOWEVENT && e.window.event == SDL_WINDOWEVENT_SIZE_CHANGED && renderer) { draw_image(); }
                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
viewflif.c: In function ‘progressive_render’:
viewflif.c:122:5: warning: implicit declaration of function ‘SDL_SetWindowTitle’ [-Wimplicit-function-declaration]
     SDL_SetWindowTitle(window,title);
     ^~~~~~~~~~~~~~~~~~
viewflif.c:124:23: error: request for member ‘w’ in something not a structure or union
       int ww = (w > dm.w ? dm.w : w);
                       ^
viewflif.c:124:30: error: request for member ‘w’ in something not a structure or union
       int ww = (w > dm.w ? dm.w : w);
                              ^
viewflif.c:125:23: error: request for member ‘h’ in something not a structure or union
       int wh = (h > dm.h ? dm.h : h);
                       ^
viewflif.c:125:30: error: request for member ‘h’ in something not a structure or union
       int wh = (h > dm.h ? dm.h : h);
                              ^
viewflif.c:128:17: error: request for member ‘w’ in something not a structure or union
       if (w > dm.w*8/10 && h > dm.h*8/10) { ww = ww*8/10; wh = wh*8/10; }
                 ^
viewflif.c:128:34: error: request for member ‘h’ in something not a structure or union
       if (w > dm.w*8/10 && h > dm.h*8/10) { ww = ww*8/10; wh = wh*8/10; }
                                  ^
viewflif.c:129:7: warning: implicit declaration of function ‘SDL_SetWindowSize’ [-Wimplicit-function-declaration]
       SDL_SetWindowSize(window,ww,wh);
       ^~~~~~~~~~~~~~~~~
viewflif.c:130:7: warning: implicit declaration of function ‘SDL_SetWindowPosition’ [-Wimplicit-function-declaration]
       SDL_SetWindowPosition(window,SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED);
       ^~~~~~~~~~~~~~~~~~~~~
viewflif.c:130:36: error: ‘SDL_WINDOWPOS_CENTERED’ undeclared (first use in this function)
       SDL_SetWindowPosition(window,SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED);
                                    ^~~~~~~~~~~~~~~~~~~~~~
viewflif.c:131:17: error: request for member ‘w’ in something not a structure or union
       if (w > dm.w*8/10 && h > dm.h*8/10) { SDL_MaximizeWindow(window); }
                 ^
viewflif.c:131:34: error: request for member ‘h’ in something not a structure or union
       if (w > dm.w*8/10 && h > dm.h*8/10) { SDL_MaximizeWindow(window); }
                                  ^
viewflif.c:131:45: warning: implicit declaration of function ‘SDL_MaximizeWindow’ [-Wimplicit-function-declaration]
       if (w > dm.w*8/10 && h > dm.h*8/10) { SDL_MaximizeWindow(window); }
                                             ^~~~~~~~~~~~~~~~~~
viewflif.c:136:38: error: ‘SDL_Texture’ undeclared (first use in this function)
     if (!image_frame) image_frame = (SDL_Texture**) calloc(flif_decoder_num_images(d), sizeof(FLIF_IMAGE*));
                                      ^~~~~~~~~~~
viewflif.c:136:51: error: expected expression before ‘)’ token
     if (!image_frame) image_frame = (SDL_Texture**) calloc(flif_decoder_num_images(d), sizeof(FLIF_IMAGE*));
                                                   ^
viewflif.c:165:29: warning: implicit declaration of function ‘SDL_DestroyTexture’ [-Wimplicit-function-declaration]
         if (image_frame[f]) SDL_DestroyTexture(image_frame[f]);
                             ^~~~~~~~~~~~~~~~~~
viewflif.c:167:26: warning: implicit declaration of function ‘SDL_CreateTextureFromSurface’ [-Wimplicit-function-declaration]
         image_frame[f] = SDL_CreateTextureFromSurface(renderer, tmpsurf);
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
viewflif.c:167:24: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
         image_frame[f] = SDL_CreateTextureFromSurface(renderer, tmpsurf);
                        ^
viewflif.c:169:9: warning: implicit declaration of function ‘SDL_SetTextureBlendMode’ [-Wimplicit-function-declaration]
         SDL_SetTextureBlendMode(image_frame[f],SDL_BLENDMODE_NONE);
         ^~~~~~~~~~~~~~~~~~~~~~~
viewflif.c:169:48: error: ‘SDL_BLENDMODE_NONE’ undeclared (first use in this function)
         SDL_SetTextureBlendMode(image_frame[f],SDL_BLENDMODE_NONE);
                                                ^~~~~~~~~~~~~~~~~~
viewflif.c: In function ‘decodeThread’:
viewflif.c:190:34: error: request for member ‘w’ in something not a structure or union
     flif_decoder_set_resize(d, dm.w*2, dm.h*2);   // the default is to not have a maximum size
                                  ^
viewflif.c:190:42: error: request for member ‘h’ in something not a structure or union
     flif_decoder_set_resize(d, dm.w*2, dm.h*2);   // the default is to not have a maximum size
                                          ^
viewflif.c: In function ‘main’:
viewflif.c:220:14: warning: implicit declaration of function ‘SDL_CreateWindow’ [-Wimplicit-function-declaration]
     window = SDL_CreateWindow("FLIF Viewer -- Loading...", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 200, 200, SDL_WINDOW_RESIZABLE);
              ^~~~~~~~~~~~~~~~
viewflif.c:220:60: error: ‘SDL_WINDOWPOS_UNDEFINED’ undeclared (first use in this function)
     window = SDL_CreateWindow("FLIF Viewer -- Loading...", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 200, 200, SDL_WINDOW_RESIZABLE);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~
viewflif.c:220:120: error: ‘SDL_WINDOW_RESIZABLE’ undeclared (first use in this function)
     window = SDL_CreateWindow("FLIF Viewer -- Loading...", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 200, 200, SDL_WINDOW_RESIZABLE);
                                                                                                                        ^~~~~~~~~~~~~~~~~~~~
viewflif.c:221:16: warning: implicit declaration of function ‘SDL_CreateRenderer’ [-Wimplicit-function-declaration]
     renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED);
                ^~~~~~~~~~~~~~~~~~
viewflif.c:221:47: error: ‘SDL_RENDERER_ACCELERATED’ undeclared (first use in this function)
     renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED);
                                               ^~~~~~~~~~~~~~~~~~~~~~~~
viewflif.c:222:5: warning: implicit declaration of function ‘SDL_SetRenderDrawColor’ [-Wimplicit-function-declaration]
     SDL_SetRenderDrawColor(renderer, 127, 127, 127, 255); // background color (in case aspect ratio of window doesn't match image)
     ^~~~~~~~~~~~~~~~~~~~~~
viewflif.c:225:9: warning: implicit declaration of function ‘SDL_GetWindowDisplayMode’ [-Wimplicit-function-declaration]
     if (SDL_GetWindowDisplayMode(window,&dm)) { printf("Error: SDL_GetWindowDisplayMode\n"); return 1; }
         ^~~~~~~~~~~~~~~~~~~~~~~~
viewflif.c:229:33: error: too many arguments to function ‘SDL_CreateThread’
     SDL_Thread *decode_thread = SDL_CreateThread(decodeThread,"Decode_FLIF",argv);
                                 ^~~~~~~~~~~~~~~~
In file included from /usr/include/SDL/SDL_audio.h:35:0,
                 from /usr/include/SDL/SDL.h:32,
                 from viewflif.c:13:
/usr/include/SDL/SDL_thread.h:88:38: note: declared here
 extern DECLSPEC SDL_Thread * SDLCALL SDL_CreateThread(int (SDLCALL *fn)(void *), void *data);
                                      ^~~~~~~~~~~~~~~~
viewflif.c:261:5: warning: implicit declaration of function ‘SDL_DestroyWindow’ [-Wimplicit-function-declaration]
     SDL_DestroyWindow(window);
     ^~~~~~~~~~~~~~~~~
Makefile:59: recipe for target 'viewflif' failed
make[1]: *** [viewflif] Error 1
make[1]: Leaving directory '/home/kberezin/code/FLIF/src'
Makefile:2: recipe for target 'flif' failed
make: *** [flif] Error 2
```

**Comment by jonsneyers** [comment]
Try `make flif` in src, instead of in the root. The makefile in the root dir was behaving differently from the one in src, fixed now: https://github.com/FLIF-hub/FLIF/commit/a65431fa24b18e51095f68c590e53ec05c0f9fd9

**Comment by jonsneyers** [comment]
Reopening because there might still be an issue in Windows.

**Comment by Enelar** [comment]
@jonsneyers yes, i am sorry. it just unusual having make file in the root of repo, but execute in src. Missed that line in readme.
